src/Npgsql/Internal/PgReader.cs

```diff
@@ -544,43 +550,53 @@ internal void ThrowIfStreamActive()
     }
 
     internal bool CommitHasIO(bool resuming) => Initialized && !resuming && FieldRemaining > 0;
-    internal ValueTask Commit(bool async, bool resuming)
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal void Commit(bool resuming)
     {
         if (!Initialized)
-            return new();
+            return;
 
         if (resuming)
         {
             if (!Resumable)
                 ThrowHelper.ThrowInvalidOperationException("Cannot resume a non-resumable read.");
-            return new();
+            return;
         }
 
         // We don't rely on CurrentRemaining, just to make sure we consume fully in the event of a nested scope not being disposed.
         // Also shut down any streaming, pooled arrays etc.
-        if (_requiresCleanup || FieldRemaining > 0)
-            return Slow(async);
+        if (_requiresCleanup || (!_fieldCompleted && FieldRemaining > 0))
```

Comment:
From the looks of things, we only set `_fieldCompleted` to `true`  if we've consumed everything (at which point `FieldRemaining` should be 0). If so, we can just get rid of `_fieldCompleted` check.

Reply:
These methods are so hot in profiles that I wanted to remove any and all duplicate indirection required on the success path. Given FieldRemaining depends on CumulativeReadPosition in NpgsqlReadBuffer (which itself is also a computed property) I wanted to reuse its result from EndRead.

I did remove the sign extension in CumulativeReadPosition after introducing _fieldCompleted so who knows what the impact still is. Either way I can't test it beyond - fairly unrepresentative - benchmarks and profiles on my machine until we do some perflab runs. On its own it'll probably fall into the noise range but that's the reason how it ended up like this.

Reply:
Done, also renamed to _fieldConsumed